armory/instrument/config.py

```diff
@@ -0,0 +1,454 @@
+    "apricot_patch_targeted_AP_per_class",
+    "dapricot_patch_targeted_AP_per_class",
+    "carla_od_AP_per_class",
+]
+# quanity_metrics only impacts output printing
+QUANTITY_METRICS = [
+    "object_detection_hallucinations_per_image",
+    "carla_od_hallucinations_per_image",
+]
+
+
+class ResultsLogWriter(LogWriter):
+    """
+    Logs successful results (designed for task metrics)
+    """
+
+    def __init__(
+        self,
+        adversarial=False,
+        targeted=False,
+        used_preds_as_labels=False,
+        log_level: str = "SUCCESS",
+    ):
+        super().__init__(log_level=log_level)
+        if targeted:
+            if adversarial:
+                self.wrt = "target"
+                self.task_type = "adversarial"
+            else:
+                raise ValueError("benign task cannot be targeted")
+        elif adversarial:
+            if used_preds_as_labels:
+                self.wrt = "benign predictions as"
+            else:
+                self.wrt = "ground truth"
+            self.task_type = "adversarial"
+        else:
+            self.wrt = "ground truth"
+            self.task_type = "benign"
+
+    def _write(self, name, batch, result):
+        # TODO: once metrics have also been updated, rewrite to be less error prone
+        #    E.g., if someone renames this from "benign_word_error_rate" to "benign_wer"
+        if "word_error_rate" in name:
+            if "total_word_error_rate" not in name:
+                result = metrics.get_supported_metric("total_wer")(result)
+            total, (num, denom) = result
+            f_result = f"total={total:.2%}, {num}/{denom}"
+        elif "entailment" in name:
+            if "total_entailment" not in name:
+                result = metrics.get_supported_metric("total_entailment")(result)
+            total = len(result)
+            f_result = (
+                f"contradiction: {result['contradiction']}/{total}, "
+                f"neutral: {result['neutral']}/{total}, "
+                f"entailment: {result['entailment']}/{total}"
+            )
+        elif any(m in name for m in MEAN_AP_METRICS):
+            if "input_to" in name:
+                for m in MEAN_AP_METRICS:
+                    if m in name:
+                        metric = metrics.get_supported_metric(m)
+                        result = metrics.MeanAP(metric)(result)
+                        break
+            f_result = f"{result}"
+        elif any(m in name for m in QUANTITY_METRICS):
+            # Don't include % symbol
+            f_result = f"{np.mean(result):.2}"
+        else:
+            f_result = f"{np.mean(result):.2%}"
+        log.success(
+            f"{name} on {self.task_type} examples w.r.t. {self.wrt} labels: {f_result}"
+        )
+
+
+def _task_metric(
+    name,
+    metric_kwargs,
+    use_mean=True,
+    include_benign=True,
+    include_adversarial=True,
+    include_targeted=True,
+    record_final_only=True,
```

Comment:
should `record_final_only` be passed as a kwarg to this function if the variable is being reset based on the metric name?

Reply:
It's only being overridden by the mAP metrics, which do not have a reasonable intermediate value to record. I think in the future we'll want to subclass Meter to have specific modes for Global and per-Sample metrics, but I don't think we're quite there yet.

In some ways, the `armory/instrument/config.py` file is a bridge from the old MetricsLogger to the new framework, but I think we'll want to revise it once we update `armory/utils/metrics.py`.